services/core-web/src/utils/helpers.js

```diff
@@ -86,6 +87,41 @@ export const normalizePhone = (value, previousValue) => {
 
 export const upperCase = (value) => value && value.toUpperCase();
 
+export const truncateFilename = (filename, max = 40) => {
+  if (filename.length <= max) {
+    return filename;
+  }
+
+  // String to use to represent that a string has been truncated
+  const trunc = "[...]";
+
+  // Extract the parts (name and extension) of the filename
+  const parts = /(^.+)\.([^.]+)$/.exec(filename);
+
+  // If the filename has no extension (e.g., the filename is "foo", not "foo.txt")
+  if (!parts) {
+    return `${filename.substring(0, max)}${trunc}`;
+  }
+
+  // Get the name of the filename (e.g., "foo.txt" will give "foo")
+  let name = parts[1];
```

Comment:
super super nitpicky but this could probably be done in one operation and maybe read a bit cleaner?

const name = parts[1].length > max ? `${parts[1].substring(0, max)}${trunc}` : parts[1];

Reply:
Sure, no problem!